xclim/ensembles/_partitioning.py

```diff
@@ -0,0 +1,188 @@
+from __future__ import annotations
+
+import numpy as np
+import pandas as pd
+import xarray as xr
+
+from ._filters import reverse_dict
+
+"""
+Uncertainty Partitioning
+========================
+
+This module implements methods and tools meant to partition climate projection uncertainties into different components:
+natural variability, GHG scenario and climate models.
+
+Partitioning algorithms:
+
+ - `hawkins_sutton`
+"""
+
+"""
+# References for other more recent algorithms
+
+Yip, S., Ferro, C. A. T., Stephenson, D. B., and Hawkins, E. (2011). A Simple, Coherent Framework for Partitioning
+Uncertainty in Climate Predictions. Journal of Climate 24, 17, 4634-4643, doi:10.1175/2011JCLI4085.1
+
+Northrop, P. J., & Chandler, R. E. (2014). Quantifying sources of uncertainty in projections of future climate.
+Journal of Climate, 27(23), 8793–8808, doi:10.1175/JCLI-D-14-00265.1
+
+Goldenson, N., Mauger, G., Leung, L. R., Bitz, C. M., & Rhines, A. (2018). Effects of ensemble configuration on
+estimates of regional climate uncertainties. Geophysical Research Letters, 45, 926– 934.
+https://doi.org/10.1002/2017GL076297
+
+Lehner, F., Deser, C., Maher, N., Marotzke, J., Fischer, E. M., Brunner, L., Knutti, R., and Hawkins,
+E. (2020). Partitioning climate projection uncertainty with multiple large ensembles and CMIP5/6, Earth Syst. Dynam.,
+11, 491–508, https://doi.org/10.5194/esd-11-491-2020.
+
+Evin, G., Hingray, B., Blanchet, J., Eckert, N., Morin, S., & Verfaillie, D. (2019). Partitioning Uncertainty
+Components of an Incomplete Ensemble of Climate Projections Using Data Augmentation, Journal of Climate, 32(8),
+2423-2440, https://doi.org/10.1175/JCLI-D-18-0606.1
+"""
+
+# Default dimension names
+_dims = dict(model="model", scenario="scenario", member="member")
+
+
+def hawkins_sutton(
```

Comment:
Should `uncertainty_paritioning` be mentioned here? or is Hawkins-Sutton universally known?

Reply:
There are other algos for uncertainty partitioning, so I want to leave room for them.